lib/smb/smb.rb

```diff
@@ -0,0 +1,68 @@
+require 'ruby_smb'
+module Intrigue
+  module Ident
+    module Smb
+      include Intrigue::Ident::SimpleSocket
+
+      # gives us the recog Smb matchers
+      def generate_smb_request_and_check(ip, port = 445, _debug = false)
+        # do the request (store as string and funky format bc of usage in core.. and  json conversions)
+        response_string = grab_response_smb(ip, port)
+        details = {
+          'details' => {
+            'response' => response_string
+          }
+        }
+
+        results = []
+
+        # generate the checks
+        checks = []
+        unless Intrigue::Ident::Smb::CheckFactory.checks.nil?
+          checks = Intrigue::Ident::Smb::CheckFactory.checks.map do |x|
+            x.new.generate_checks
+          end.compact.flatten
+        end
+        # and run them against our result
+        checks.each do |check|
+          results << match_smb_response_hash(check, details)
+        end
+        { 'fingerprint' => results.uniq.compact, 'response' => response_string, 'content' => [] }
+      end
+
+      private
+
+      def grab_response_smb(ip, port, _timeout = 30)
+        # if socket = connect_tcp(ip, port, timeout)
+        sock = TCPSocket.new ip, port
+        dispatcher = RubySMB::Dispatcher::Socket.new(sock)
+        client = RubySMB::Client.new(dispatcher, username: '', password: '')
+
+        out = []
+        begin
+          out << "Protocol: #{client.negotiate}"
+          client_auth = client.authenticate
+          out << "Description: #{client_auth.description}"
+          out << "Connection Status: #{client_auth.name}"
+        rescue StandardError => e
+          out = "Failed to connect to #{ip}: #{e.message}"
+        end
+
+        begin
+          out << "Native OS: #{client.peer_native_os}" if client.peer_native_os
```

Comment:
@duartemateus Any reason not to use a hash for this?

Reply:
Not any good reason no haha. I'll make the change and also create appropriate grabber functions for each key. I'll give you a nudge when it's complete, shouldn't take long

Reply:
https://github.com/intrigueio/intrigue-ident/pull/105/commits/7c4517e7bc771f6812b3c49c4a97d8e8795f450c 

@jcran

Reply:
🔥